keras_nlp/src/models/retinanet/box_matcher.py

```diff
@@ -0,0 +1,265 @@
+       positive_threshold; the rows will be assigned to ignore_value if its
+       argmax result < negative_threshold, and the rows will be assigned to
+       negative_value if its argmax result is between [negative_threshold,
+       positive_threshold). This is different from case 1) by swapping first two
+       values.
+    3) thresholds=[positive_threshold], and
+       match_values=[negative_values, positive_value]: the rows will be assigned
+       to positive value if its argmax result >= positive_threshold; the rows
+       will be assigned to negative_value if its argmax result <
+       negative_threshold.
+
+    Args:
+        thresholds: A sorted list of floats to classify the matches into
+          different results (e.g. positive or negative or ignored match). The
+          list will be prepended with -Inf and and appended with +Inf.
+        match_values: A list of integers representing matched results (e.g.
+          positive or negative or ignored match). len(`match_values`) must
+          equal to len(`thresholds`) + 1.
+        force_match_for_each_col: each row will be argmax matched to at
+          least one column. This means some columns will be matched to
+          multiple rows while some columns will not be matched to any rows.
+          Filtering by `thresholds` will make less columns match to positive
+          result. Setting this to True guarantees that each column will be
+          matched to positive result to at least one row.
+
+    Raises:
+        ValueError: if `thresholds` not sorted or
+        len(`match_values`) != len(`thresholds`) + 1
+
+    Example:
+
+    ```python
+    box_matcher = keras_cv.layers.BoxMatcher([0.3, 0.7], [-1, 0, 1])
+    iou_metric = keras_cv.bounding_box.compute_iou(anchors, boxes)
+    matched_columns, matched_match_values = box_matcher(iou_metric)
+    cls_mask = ops.less_equal(matched_match_values, 0)
+    ```
+
+    """
+
+    def __init__(
+        self,
+        thresholds: List[float],
+        match_values: List[int],
+        force_match_for_each_col: bool = False,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if sorted(thresholds) != thresholds:
+            raise ValueError(f"`threshold` must be sorted, got {thresholds}")
+        self.match_values = match_values
+        if len(match_values) != len(thresholds) + 1:
+            raise ValueError(
+                f"len(`match_values`) must be len(`thresholds`) + 1, got "
+                f"match_values {match_values}, thresholds {thresholds}"
+            )
+        thresholds.insert(0, -float("inf"))
+        thresholds.append(float("inf"))
+        self.thresholds = thresholds
+        self.force_match_for_each_col = force_match_for_each_col
+        self.built = True
+
+    def call(self, similarity_matrix):
+        """Matches each row to a column based on argmax
+
+        Args:
+          similarity_matrix: A float Tensor of shape [num_rows, num_cols] or
```

Comment:
four spaces indent for Args:, Returns:, etc

Reply:
Done